mlflow/sklearn/utils.py

```diff
@@ -26,12 +28,18 @@
 # on scikit-learn older than this version.
 _MIN_SKLEARN_VERSION = "0.20.3"
 
-_METRICS_PREFIX = "training_"
+# The prefix to note that all calculated metrics and artifacts are solely based on training datasets
+_TRAINING_PREFIX = "training_"
+
 _SAMPLE_WEIGHT = "sample_weight"
 
-# _SklearnMetric represents a metric (e.g, precision_score) that will be computed and logged
-# during the autologging routine for a particular model type (eg, classifier, regressor).
-_SklearnMetric = collections.namedtuple("_SklearnMetric", ["name", "function", "arguments"])
+# _SklearnMetricOrArtifact represents a metric (e.g, precision_score)
+# or artifact (e.g confusion matrix) that will be computed and
+# logged during the autologging routine for a particular model type
+# (eg, classifier, regressor). "title" will only be used for recording artifacts
+_SklearnMetricOrArtifact = collections.namedtuple(
```

Comment:
For readability and potential extensibility later, I think I'd rather maintain separate `_SklearnMetric` and `_SklearnArtifact` named tuple objects - especially because they contain different required fields (artifacts all specify a non-empty title) and are consumed by different routines that explicitly assume the object is an artifact or a metric. Can we split these up?